backend/httpclient/http_client_test.go

```diff
@@ -10,13 +10,21 @@ import (
 )
 
 func TestNewClient(t *testing.T) {
-	t.Run("New() without any opts should return http.DefaultClient", func(t *testing.T) {
+	t.Run("New() without any opts should return expected http client and middlewares", func(t *testing.T) {
 		client, err := New()
 		require.NoError(t, err)
-		require.Same(t, http.DefaultClient, client)
+		require.NotNil(t, client)
+		require.Equal(t, 30*time.Second, client.Timeout)
```

Comment:
Would also adding 

```go
require.NotSame(t, http.DefaultClient, client)
```

show clearer intention?

Reply:
certainly wouldn't hurt, I'll add that